app/views/admin/categories/index.html.erb

```diff
@@ -24,10 +24,49 @@
     <div class="col-sm-12 col-md-4 add-category__btn" style= "height: 450px;">
       <div class="container">
         <!-- Trigger the modal with a button -->
-        <button type="button" class="btn btn-info btn-lg" data-toggle="modal" data-target="#myModal"><i class="fa fa-plus"></i> </button>
+        <button type="button" class="btn btn-info btn-lg btn__add" data-toggle="modal" data-target="#newCategory"><i class="fa fa-plus"></i> </button>
+        <div class="modal fade" id="newCategory" role="dialog">
+          <div class="modal-dialog">
+            <div class="modal-content">
+              <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">&times;</button>
+                <h4 class="modal-title"><%=t "admin.categories.create_title"%></h4>
+              </div>
+              <div class="modal-body">
+                <div class = "container">
+                  <%= form_for(@category, url: admin_categories_path) do |form| %>
+                    <div class = "row">
+                      <div class = "col-md-3">
+                        <div class = "row"><%= form.text_field :name, :placeholder => t("admin.categories.name"),
+                          style: "font-size: 20px;margin-bottom: 15px;height: 30px" %></div>
+                        <div class = "row">
+                          <label for="exampleInputFile"><%=t "admin.categories.add_image"%></label>
+                          <%= form.file_field :image, type:"file", onchange: "loadFile(event)"%>
+                        </div>
+                      </div>
+                      <div class="preview col-md-3">
+                        <img id = "output" style = "height: 100px" />
+                      </div>
+                    </div>
+                </div>
+              </div>
+              <div class="modal-footer">
+                  <%= form.submit t "admin.categories.save", class:"btn-save", type:"submit" %>
+                  <% end %>
```

Comment:
Nút end này là đóng thẻ của thẻ nào ri em ?

Reply:
dạ đóng của form for a